File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/actions/fight/GameActionFightCastOnTargetRequestMessage.scala





















// Generated on 05/03/2013 10:34:32
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait GameActionFightCastOnTargetRequestMessage extends MessageDeserializer {
  val messageId = 6330

  protected trait Message extends BaseMessage with Serializable {
    val messageId = GameActionFightCastOnTargetRequestMessage.messageId

    val spellId: Short
    val targetId: Int

    def serialize(buf: Buffer) {
      buf.writeShort(spellId)
      buf.writeInt(targetId)
    }

  }

  protected def create(spellId: Short, targetId: Int): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val spellId = buf.readShort()
    val targetId = buf.readInt()

    create(spellId, targetId)
  }

}

object GameActionFightCastOnTargetRequestMessage extends GameActionFightCastOnTargetRequestMessage {
  case class MessageImpl(spellId: Short, targetId: Int) extends Message
  type Target = MessageImpl

  override def create(spellId: Short, targetId: Int) = MessageImpl(spellId, targetId)

  def apply(spellId: Short, targetId: Int) = create(spellId, targetId)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/GameActionFightCastOnTargetRequestMessage.java


// Generated on 05/08/2013 19:37:38
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class GameActionFightCastOnTargetRequestMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6330;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public short spellId;
    public int targetId;
    
    public GameActionFightCastOnTargetRequestMessage() { }
    
    public GameActionFightCastOnTargetRequestMessage(short spellId, int targetId) {
        this.spellId = spellId;
        this.targetId = targetId;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeShort(spellId);
        buf.writeInt(targetId);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        spellId = buf.readShort();
        if (spellId < 0)
            throw new RuntimeException("Forbidden value on spellId = " + spellId + ", it doesn't respect the following condition : spellId < 0");
        targetId = buf.readInt();
    }
    
}
